crates/sui-core/src/execution_scheduler/balance_withdraw_scheduler/object_balance/naive_scheduler.rs

```diff
@@ -0,0 +1,211 @@
+            accumulator_version_receiver: Arc::new(accumulator_version_receiver),
+        }
+    }
+
+    pub fn schedule(
+        &self,
+        object_withdraws: BTreeMap<AccumulatorObjId, u64>,
+        accumulator_version: SequenceNumber,
+    ) -> ObjectBalanceWithdrawStatus {
+        let last_settled_version = *self.accumulator_version_receiver.borrow();
+        // This function is called during execution, which means this transaction is not committed yet,
+        // so the settlement transaction at the end of the same consensus commit cannot have settled yet.
+        assert!(accumulator_version >= last_settled_version);
+        // TODO: Should we use last_settled_version or accumulator_version here?
+        if self.check_balance_sufficient(&object_withdraws, accumulator_version) {
+            self.record_withdraws(&object_withdraws, accumulator_version);
+            return ObjectBalanceWithdrawStatus::SufficientBalance;
+        }
+        if accumulator_version == last_settled_version {
+            return Self::return_insufficient_balance();
+        }
+
+        // Spawn a task to wait for the last settled version to become accumulator_version,
+        // before we could check again.
+        let scheduler = self.clone();
+        let accumulator_version_sender = self.accumulator_version_sender.clone();
+        let (sender, receiver) = oneshot::channel();
+        tokio::spawn(async move {
+            let mut version_receiver = accumulator_version_sender.subscribe();
+            version_receiver
+                .wait_for(|v| *v == accumulator_version)
+                .await
+                .unwrap();
+            // TODO: Is this unwrap safe?
+
+            if scheduler.check_balance_sufficient(&object_withdraws, accumulator_version) {
+                // TODO: Is this unwrap safe?
+                sender.send(BalanceWithdrawStatus::Unknown).unwrap();
+            } else {
+                // TODO: Is this unwrap safe?
+                sender
+                    .send(BalanceWithdrawStatus::InsufficientBalance)
+                    .unwrap();
+            }
+        });
+        ObjectBalanceWithdrawStatus::Pending(receiver)
+    }
+
+    pub fn settle_accumulator_version(&self, next_accumulator_version: SequenceNumber) {
+        let mut inner_state = self.inner_state.lock();
+        while let Some(version) = inner_state.tracked_accounts.keys().next().copied() {
+            let accounts = if version < next_accumulator_version {
+                inner_state.tracked_accounts.remove(&version).unwrap()
+            } else {
+                break;
+            };
+            for obj_id in accounts {
+                let unsettled = inner_state.unsettled_withdraws.get_mut(&obj_id).unwrap();
+                let version_amount = unsettled.unsettled_withdraw_map.remove(&version).unwrap();
+                unsettled.total_amount -= version_amount;
+            }
+        }
+
+        // TODO: Is this unwrap safe?
```

Comment:
it is not, since there may not be any listeners: https://docs.rs/tokio/latest/tokio/sync/watch/struct.Sender.html#method.send

you can use one of the other send methods, probably `send_replace` is the best one here